api/app/playwright_renderer.py

```diff
@@ -0,0 +1,112 @@
+import os
+from typing import Final, Optional
+from urllib.parse import parse_qs, urlparse
+
+from app.caching import CACHE_DIRECTORY
+from playwright.async_api import async_playwright
+
+# Html selectors
+LAYER_ACCORDION_SELECTOR: Final[
+    str
+] = 'div[class="MuiAccordionSummary-root"], div[aria-expanded="false"]'
+DOWNLOAD_BUTTON_SELECTOR: Final[str] = "a[download]"
+CREATE_REPORT_BUTTON_SELECTOR: Final[str] = 'button[id="create-report"]'
+
+# Timeouts
+PAGE_TIMEOUT: Final[int] = 25000
+PAGE_LANGUAGE_CHANGE_TIMEOUT: Final[int] = 5000
+
+
+async def playwright_download_report(url: str, language: Optional[str]) -> str:
+    language = "en" if language is None else language
+    layerIdParam = extract_query_param(url, "hazardLayerIds")
+    dateParam = extract_query_param(url, "date")
+    report_file_path = (
+        CACHE_DIRECTORY
+        + "reports/report-"
+        + layerIdParam
+        + "-"
+        + language
+        + "-"
+        + dateParam
+        + ".pdf"
+    )
+
+    if os.path.exists(report_file_path):
+        return report_file_path
+
+    async with async_playwright() as p:
+        browser = await p.chromium.launch()
+        page = await browser.new_page()
+
+        page.set_default_timeout(PAGE_TIMEOUT)
+        await page.goto(url)
+
+        # Wait for page to be loaded
+        await page.wait_for_selector(LAYER_ACCORDION_SELECTOR, state="visible")
+
+        # Toggle level one dropdowns
+        await toggle_every_visible_dropdown(page)
+
+        # Toggle level two dropdowns
+        await toggle_every_visible_dropdown(page)
+
+        # Click on exposure analysis toggle button
+        await click_target_exposure_analysis(page, layerIdParam)
+
+        # Wait for page to be loaded on exposure analysis
+        await page.wait_for_selector(CREATE_REPORT_BUTTON_SELECTOR, state="visible")
+
+        # Change language if not english
+        await change_language_if_not_default(language, page)
```

Comment:
Some functions have page as the first argument while others have page as the second. Let's try to be consistent to avoid confusions